src/wallet/rpcwallet.cpp

```diff
@@ -2298,6 +2298,17 @@ UniValue walletpassphrase(const JSONRPCRequest& request)
     // Alternately, find a way to make request.params[0] mlock()'d to begin with.
     strWalletPass = request.params[0].get_str().c_str();
 
+    // Get the timeout
+    int64_t nSleepTime = request.params[1].get_int64();
```

Comment:
What was the root cause of  the behaviour described in #12100?
Is it not this line that is causing this strange behaviour?

Reply:
The root cause is that the int64 can have a higher value than what `time_t` can store. This gets around that problem by making `nSleepTime` at most a specific value that is known to be one that `time_t` can store.